### PR TITLE
Feat(interceptor): add transformer interceptor

### PR DESCRIPTION
**Pull request by ethfoo**
#### Proposed Changes:

* add transformer interceptor
* conditions: equal(), contain(), hasPrefix, exist(), match(), oneof(), greater(), less()
* actions: add(), set(), copy(), del(), fmt(), jsondecode(), move(), regex(), return(), strconv(), underRoot(), print(), timestamp()

#### Which issue(s) this PR fixes:

None

#### Additional documentation:

```yaml
- interceptors:
  - type: transformer
    actions:
    - if: <conditions>
      then:
      - action: <action>
        <other options>
      - action: <action>
      else:
      - action: <action>
    - if: <conditions>
      then:
      - action: <action>      
    - action: <action>
```

### CODE REVIEW

**Review by iamyeka — CHANGES_REQUESTED** [review]

File: pkg/interceptor/transformer/condition/oneof.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+package condition
+
+import (
+	"github.com/loggie-io/loggie/pkg/core/api"
+	"github.com/loggie-io/loggie/pkg/util/eventops"
+	"github.com/pkg/errors"
+)
+
+const (
+	OneOfName     = "oneof"
```
[the remark this answers is not in the document]
Comment (ethfoo) [reply pkg/interceptor/transformer/condition/oneof.go:26]: ok

File: pkg/interceptor/transformer/condition/greater.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+		return nil, err
+	}
+
+	return &Greater{
+		field: args[0],
+		value: number,
+	}, nil
+}
+
+func (eq *Greater) Check(e api.Event) bool {
```
[the remark this answers is not in the document]
Comment (ethfoo) [reply pkg/interceptor/transformer/condition/greater.go:59]: done

File: pkg/interceptor/transformer/condition/isnull.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+import (
+	"github.com/loggie-io/loggie/pkg/core/api"
+	"github.com/loggie-io/loggie/pkg/util/eventops"
+	"github.com/pkg/errors"
+)
+
+const (
+	IsNullName     = "isNull"
+	IsNullUsageMsg = "usage: isNull(key)"
```
[the remark this answers is not in the document]
Comment (ethfoo) [reply pkg/interceptor/transformer/condition/isnull.go:27]: done

File: pkg/interceptor/transformer/action/dropevent.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+import (
+	"errors"
+	"fmt"
+	"github.com/loggie-io/loggie/pkg/core/api"
+	"github.com/loggie-io/loggie/pkg/core/cfg"
+)
+
+const (
+	DropEventName     = "dropEvent"
```
[the remark this answers is not in the document]
Comment (ethfoo) [reply pkg/interceptor/transformer/action/dropevent.go:27]: 因为drop再normalize里面表示删除一个字段，所以这里为了不让用户混淆，强调了一下是dropEvent

**Pull request merged**